### PR TITLE
Added kselect option to test runner

### PR DESCRIPTION
**Pull request by eteq**
I got tired of constantly doing `python setup.py test --args="-k <something>"` to select particular tests by name, so this adds a command to the astropy test runner that does this directly (without needing to go through `args`)

### CODE REVIEW

File: astropy/__init__.py [hunk 1]
```diff
@@ -189,6 +189,9 @@ def test(package=None, test_path=None, args=None, plugins=None,
     skip_docs : bool, optional
         When `True`, skips running the doctests in the .rst files.
 
+    kselect : str, optional
+        Test selection, passed to py.test as "-k <kselect>".
```
Comment (embray) [inline astropy/__init__.py]: `-k` is fine for the short name.   In principle we don't even need a long name, but if we had one I'd just say `--test-name` or something like that.  The point is that it matches the test name against a pattern.

**Comment by eteq** [comment]
Note that this probably needs a major overhaul following #4020 .

**Comment by embray** [comment]
Just a bit of shuffling around really.  I can even do it if you want.

**Comment by eteq** [comment]
I'm just going to close this as it would be better to start afresh at this point anyway, whether or not this is worth doing.

**Pull request closed without merging**